scripts/sbatch.py

```diff
@@ -0,0 +1,288 @@
+
+import sys
+import os
+import json
+import subprocess
+import argparse
+import getpass
+from enum import Enum
+
+MB_PER_GB = 1024.0
+
+DEFAULT_NTASKS = 1
+DEFAULT_CPUS_PER_TASK = 1
+
+MAHUIKA_SLURM_CONF = (
+    "/scale_wlg_persistent/filesets/home/slurm/mahuika/etc/opt/slurm/slurm.conf"
+)
+MAUI_SLUM_CONF = "/etc/opt/slurm.conf"
+
+SBATCH_BIN = "/opt/slurm/18.08.7/bin/sbatch"
+
+
+# can use from qcore
+class HPC(Enum):
+    maui = "maui"
+    mahuika = "mahuika"
+
+
+# can be moved to qcore const
+class SlurmHeader(Enum):
+    account = "account"
+    partition = "partition"
+    ntasks = "ntasks"
+    cpus_per_task = "cpus-per-task"
+    mem_per_cpu = "mem-per-cpu"
+
+
+class MahuikaPartition(Enum):
+    large = "large"
+    long = "long"
+    perpost = "prepost"
+    bigmem = "bigmem"
+    hugmem = "hugemem"
+    ga_bigmem = "ga_bigmem"
+    ga_hugemem = "ga_hugemem"
+    gpu = "gpu"
+    igpu = "igpu"
+
+
+class MauiPartition(Enum):
+    nesi_research = "nesi_research"
+
+
+def get_billing_weights(line):
+    """Get cpu_weights and memory_weights from the line containing TresBilling attribute on mahuika"""
+    # ["'CPU=1.0,Mem=0.1429G'", 'QOS=p_prepost2', 'MaxTime=03:00:00', 'PriorityTier=1']
+    cpu, mem = line.strip().split("TRESBillingWeights=")[-1].split()[0].split(",")
+    cpu = float(cpu.split("=")[-1])
+    mem = float(mem.split("=")[-1][:-2])
+
+    return cpu, mem
+
+
+def get_default_mem_per_cpu(slurm_conf):
+    """"Read a slurm.conf file and gets the default memory per cpu"""
+    with open(slurm_conf, "r") as f:
+        lines = f.readlines()
+    # PartitionName=DEFAULT PreemptMode=OFF DefMemPerCPU=512 DefaultTime=15 MaxTime=3-00:00 TRESBillingWeights="CPU=0.5,Mem=0.3333G"
+    for line in lines[::-1]:
+        if line.startswith("PartitionName=DEFAULT"):
+            # 512 /1024
+            default_mem_per_cpu = (
+                float(line.strip().split("DefMemPerCPU=")[-1].split()[0]) / MB_PER_GB
+            )
+            return default_mem_per_cpu
+
+
+def read_slurm_conf(slurm_conf, partition_name):
+    """
+       Read a slurm.conf and returns the correspoding cpu_weights and mem_weights according to the partition_name
+       Currently only used for mahuika slurm.conf
+    """
+    with open(slurm_conf, "r") as f:
+        lines = f.readlines()
+    # PartitionName is at the end of file
+    for line in lines[::-1]:
+        if line.startswith("PartitionName={}".format(partition_name)):
+            # PartitionName=prepost    Nodes=wbl[001-005,008-011] TRESBillingWeights="CPU=1.0,Mem=0.1429G"  QOS=p_prepost2 MaxTime=03:00:00 PriorityTier=1
+            try:
+                return get_billing_weights(line)
+            except (IndexError, ValueError):
+                # TRESBillingWeights not found, use default cpu_weights and mem_weights
+                for l in lines[::-1]:
+                    if l.startswith("PartitionName=DEFAULT"):
+                        return get_billing_weights(l)
+
```

Comment:
set if BillingWeight is not even set in partition=DEFAULT, then weight_cpu=1.0 weight_mem=0
so Maui will not needed to be treated as a special case.